src/symbol/grid_index.js

```diff
@@ -117,17 +117,20 @@ class GridIndex {
                     y2: y + radius
                 });
             }
+            if (predicate) {
+                return result.filter(predicate);
+            }
         } else {
             const queryArgs = {
                 hitTest,
                 seenUids: { box: {}, circle: {} }
             };
-            this._forEachCell(x1, y1, x2, y2, this._queryCell, result, queryArgs);
+            this._forEachCell(x1, y1, x2, y2, this._queryCell, result, queryArgs, predicate);
         }
         return hitTest ? result.length > 0 : result;
```

Comment:
Can this return now be moved into the `else` block?

Reply:
Yeah, that might be a cleaner way to write it, but we need to make sure to return something in the first clause if `predicate` is undefined.

Reply:
Updated.